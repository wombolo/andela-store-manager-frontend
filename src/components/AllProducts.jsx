import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getAllProducts, addToCart } from '../actions/productActions';
import {Link} from "react-router-dom";

export class AllProducts extends Component {
  componentWillMount() {
    this.props.getAllProducts();
  }

  addToCartEvt = (product) =>{
    this.props.addToCart(product);
  };

  render() {
    const {products} = this.props.products;

    if (products.length < 1){
      return(<p>Now Loading. Please be patient</p>)
    }


    return (
      <div>
        <h1>All Products</h1>

        <div className="row big-row products-row">
          {products.map(product => {
            const imageFile = require(`../assets/images/${product.image}`);

              return (
                <div key={product.id}>
                  <figure className="image-holder">
                    <Link to={`/edit-product/${product.id}`}>
                      <img src={imageFile} />
                    </Link>
                  </figure>
                  <div className="row">
                    <div className="col-3"><span className="pull-left item-price"> $ {product.price}</span></div>
                    <div className="col-6">

                      <button className="pull-right min-btn btn-primary" onClick={() => this.addToCartEvt(product)}>Add to Cart</button>
                    </div>
                  </div>
                  <h3>
                    <Link to={`/edit-product/${product.id}`}>
                    {product.title}</Link>
                  </h3>
                </div>
              )
            })
          }
        </div>

        <div className="row m-t-5 text-center">
          <h3 className='text-gold'>You may Scroll Right or Left</h3>
        </div>
        </div>
    )
  }
}

AllProducts.propTypes = {
  products: PropTypes.shape({}).isRequired,
  getAllProducts: PropTypes.func.isRequired,
  addToCart: PropTypes.func.isRequired
};

export const mapStateToProps = state => ({
  products: state.products,
});

export default connect(mapStateToProps, {
  getAllProducts,addToCart
})(AllProducts);

