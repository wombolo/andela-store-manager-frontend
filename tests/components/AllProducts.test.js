import 'babel-polyfill'
import React from 'react';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';
import { MemoryRouter } from 'react-router-dom';


import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

import { AllProducts } from '../../src/components/AllProducts.jsx';

let wrapper;

const props = {
  getAllProducts: () => Promise.resolve(),
  addToCart: () => Promise.resolve(),
  products: {
    products: [
      {id: 5, title: "Bonjour", image: "products/product-5.png",price: "150.00", quantity: 3, status: "active"},
      {id: 2, title: "Bonjour", image: "products/product-5.png",price: "150.00", quantity: 3, status: "active"},
    ]
  }
};


const mockStore = configureStore([thunk]);

const store = mockStore({ ...props });




describe('AllProducts Component', () => {
  it('should match snapshot', () => {
    wrapper = shallow(<AllProducts {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });

  describe('Methods in Class', () => {
    beforeEach(() => {
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter>
          <AllProducts {...props} />
          </MemoryRouter>
        </Provider>,
      );
    });

    afterEach(() => {
      wrapper.unmount();
    });

    it('triggers addToCart()', () => {
      const AllProductsComp = wrapper.find('AllProducts')
      sinon.spy(AllProductsComp.instance(), 'addToCartEvt');

      AllProductsComp.instance().addToCartEvt(props.products.products[0]);
      expect(AllProductsComp.instance().addToCartEvt.calledOnce)
        .toEqual(true);

      const button = AllProductsComp.find('button').last();
      button.simulate('click', props.products.products[0]);
    });

  });
});
