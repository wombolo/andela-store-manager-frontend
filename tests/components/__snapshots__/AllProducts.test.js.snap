// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllProducts Component should match snapshot 1`] = `
<div>
  <h1>
    All Products
  </h1>
  <div
    className="row big-row products-row"
  >
    <div
      key="5"
    >
      <figure
        className="image-holder"
      >
        <Link
          replace={false}
          to="/edit-product/5"
        >
          <img
            src=""
          />
        </Link>
      </figure>
      <div
        className="row"
      >
        <div
          className="col-3"
        >
          <span
            className="pull-left item-price"
          >
             $ 
            150.00
          </span>
        </div>
        <div
          className="col-6"
        >
          <button
            className="pull-right min-btn btn-primary"
            onClick={[Function]}
          >
            Add to Cart
          </button>
        </div>
      </div>
      <h3>
        <Link
          replace={false}
          to="/edit-product/5"
        >
          Bonjour
        </Link>
      </h3>
    </div>
    <div
      key="2"
    >
      <figure
        className="image-holder"
      >
        <Link
          replace={false}
          to="/edit-product/2"
        >
          <img
            src=""
          />
        </Link>
      </figure>
      <div
        className="row"
      >
        <div
          className="col-3"
        >
          <span
            className="pull-left item-price"
          >
             $ 
            150.00
          </span>
        </div>
        <div
          className="col-6"
        >
          <button
            className="pull-right min-btn btn-primary"
            onClick={[Function]}
          >
            Add to Cart
          </button>
        </div>
      </div>
      <h3>
        <Link
          replace={false}
          to="/edit-product/2"
        >
          Bonjour
        </Link>
      </h3>
    </div>
  </div>
  <div
    className="row m-t-5 text-center"
  >
    <h3
      className="text-gold"
    >
      You may Scroll Right or Left
    </h3>
  </div>
</div>
`;
